frontend/src/Components/EnrollmentsTable.tsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import DropdownControl from '@/Components/inputs/DropdownControl';
+import { ClassEnrollment } from '@/common';
+
+interface EnrollmentTableProps {
+    enrollments: ClassEnrollment[];
+    inlineOptions: Record<string, string>;
+    selectedResidents: number[];
+    toggleSelection: (userId: number) => void;
+    handleSelectAll: () => void;
+    areAllSelected: boolean;
+    canToggle: (enrollment: ClassEnrollment) => boolean;
+    isEditable: (enrollment: ClassEnrollment) => boolean;
+    handleChange: (value: string, enrollment: ClassEnrollment) => void;
+    handleShowCompletionDetails: (enrollment: ClassEnrollment) => Promise<void>;
+}
+
+const EnrollmentsTable: React.FC<EnrollmentTableProps> = ({
+    enrollments,
+    inlineOptions,
+    selectedResidents,
+    toggleSelection,
+    handleSelectAll,
+    areAllSelected,
+    canToggle,
+    isEditable,
+    handleChange,
+    handleShowCompletionDetails
+}) => {
+    return (
+        <div className="relative w-full" style={{ overflowX: 'clip' }}>
+            <table className="table-2 mb-4">
+                <thead>
+                    <tr className="grid grid-cols-6 pb-4">
+                        <th className="justify-self-start pl-4">
+                            <input
+                                className="checkbox"
+                                type="checkbox"
+                                checked={areAllSelected}
+                                onChange={handleSelectAll}
+                            />
+                        </th>
+                        <th>Name</th>
+                        <th>DOC ID</th>
+                        <th>Enrolled Date</th>
+                        <th>Completion</th>
+                        <th className="justify-self-end pr-4">
+                            Enrollment Status
+                        </th>
+                    </tr>
+                </thead>
+                <tbody>
+                    {enrollments.length > 0 ? (
+                        enrollments.map((enrollment: ClassEnrollment) => (
+                            <tr
+                                key={enrollment.id}
+                                className={`card h-16 w-full grid-cols-6 justify-items-center cursor-pointer ${!canToggle(enrollment) ? 'bg-gray-100' : ''}`}
```

Comment:
@calisio Is definitely going to notice the usage of `bg=-gray-100` here. We should probably use one of the themed colors when something is selected instead. I don't see any other examples that immediately come to mind in the app though that we can reference.